### DISCUSSION TITLE
Advice needed - Error: Process completed with exit code 1

### DISCUSSION THREAD

**Discussion by SimonSaysTy** [nickcoutsos/keymap-editor#234] [Q&A] [answered]
Hello all,  

I have been repeatably running into this issue when I am compiling a new keymap and I am not sure why. Whenever it is compiling, it will run everything smoothly but when it gets to the west build segment it fails every time.  This is the error it is giving me;

> Run west build -s zmk/app -d "/tmp/tmp.y64XXKNuXN" -b "nice_nano_v2"  -- -DZMK_CONFIG=/__w/zmk-config-klorv2/zmk-config-klorv2/config -DSHIELD="klor_right" 
> + west build -s zmk/app -d /tmp/tmp.y64XXKNuXN -b nice_nano_v2 -- -DZMK_CONFIG=/__w/zmk-config-klorv2/zmk-config-klorv2/config -DSHIELD=klor_right
> -- west build: generating a build system
> Loading Zephyr default modules (Zephyr base).
> -- Application: /__w/zmk-config-klorv2/zmk-config-klorv2/zmk/app
> -- CMake version: 3.28.1
> -- Found Python3: /usr/bin/python3 (found suitable version "3.8.10", minimum required is "3.8") found components: Interpreter 
> -- Cache files will be written to: /__w/zmk-config-klorv2/zmk-config-klorv2/zephyr/.cache
> -- Zephyr version: 3.5.0 (/__w/zmk-config-klorv2/zmk-config-klorv2/zephyr)
> -- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
> -- Adding ZMK config directory as board root: /__w/zmk-config-klorv2/zmk-config-klorv2/config
> -- Adding /__w/zmk-config-klorv2/zmk-config-klorv2/config/boards/shields/klor
> -- ZMK Config directory: /__w/zmk-config-klorv2/zmk-config-klorv2/config
> -- ZMK Config Kconfig: /__w/zmk-config-klorv2/zmk-config-klorv2/config/klor.conf
> -- Using keymap file: /__w/zmk-config-klorv2/zmk-config-klorv2/config/klor.keymap
> -- Using keymap file: /__w/zmk-config-klorv2/zmk-config-klorv2/config/klor.keymap
> -- Board: nice_nano_v2
> -- Shield(s): klor_right
> -- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
> -- Found host-tools: zephyr 0.16.3 (/opt/zephyr-sdk-0.16.3)
> -- Found toolchain: zephyr 0.16.3 (/opt/zephyr-sdk-0.16.3)
> -- Found Dtc: /opt/zephyr-sdk-0.16.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
> -- Found BOARD.dts: /__w/zmk-config-klorv2/zmk-config-klorv2/zmk/app/boards/arm/nice_nano/nice_nano_v2.dts
> -- Found devicetree overlay: /__w/zmk-config-klorv2/zmk-config-klorv2/config/boards/shields/klor/klor_right.overlay
> -- Found devicetree overlay: /__w/zmk-config-klorv2/zmk-config-klorv2/config/boards/shields/klor/boards/nice_nano_v2.overlay
> -- Found devicetree overlay: /__w/zmk-config-klorv2/zmk-config-klorv2/config/klor.keymap
> devicetree error: Label 'spi1_default' appears on /pin-controller/spi1_default and on /pin-controller/spi3_default
> CMake Error at /__w/zmk-config-klorv2/zmk-config-klorv2/zephyr/cmake/modules/dts.cmake:279 (message):
>   gen_defines.py failed with return code: 1
> -- In: /tmp/tmp.y64XXKNuXN/zephyr, command: /usr/bin/python3;/__w/zmk-config-klorv2/zmk-config-klorv2/zephyr/scripts/dts/gen_defines.py;--dts;/tmp/tmp.y64XXKNuXN/zephyr/zephyr.dts.pre;--dtc-flags;'';--bindings-dirs;/__w/zmk-config-klorv2/zmk-config-klorv2/zmk/app/module/dts/bindings;/__w/zmk-config-klorv2/zmk-config-klorv2/zmk/app/dts/bindings;/__w/zmk-config-klorv2/zmk-config-klorv2/zephyr/dts/bindings;--header-out;/tmp/tmp.y64XXKNuXN/zephyr/include/generated/devicetree_generated.h.new;--dts-out;/tmp/tmp.y64XXKNuXN/zephyr/zephyr.dts.new;--edt-pickle-out;/tmp/tmp.y64XXKNuXN/zephyr/edt.pickle;--vendor-prefixes;/__w/zmk-config-klorv2/zmk-config-klorv2/zmk/app/dts/bindings/vendor-prefixes.txt;--vendor-prefixes;/__w/zmk-config-klorv2/zmk-config-klorv2/zephyr/dts/bindings/vendor-prefixes.txt
> Call Stack (most recent call first):
>   /__w/zmk-config-klorv2/zmk-config-klorv2/zephyr/cmake/modules/zephyr_default.cmake:[12](https://github.com/SimonSaysTy/zmk-config-klorv2/actions/runs/9669326455/job/26675556959#step:11:13)9 (include)
>   /__w/zmk-config-klorv2/zmk-config-klorv2/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
>   /__w/zmk-config-klorv2/zmk-config-klorv2/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
>   CMakeLists.txt:8 (find_package)
> 
> 
> -- Configuring incomplete, errors occurred!
> FATAL ERROR: command exited with status 1: /usr/local/bin/cmake -DWEST_PYTHON=/usr/bin/python3 -B/tmp/tmp.y64XXKNuXN -GNinja -DBOARD=nice_nano_v2 -DZMK_CONFIG=/__w/zmk-config-klorv2/zmk-config-klorv2/config -DSHIELD=klor_right -S/__w/zmk-config-klorv2/zmk-config-klorv2/zmk/app
> Error: Process completed with exit code 1.

I have not changed anything else in my ZMK file, and have only ever edited my keymap using this editor. Has anyone else run into an issue like this before? How did you solve it?

**Comment by nickcoutsos** [comment, marked as answer]
This is kind of out of the scope of keymap editor, but from a cursory glance it seems that

- the relevant error output is:
    > devicetree error: Label 'spi1_default' appears on /pin-controller/spi1_default and on /pin-controller/spi3_default
- [this commit](https://github.com/SimonSaysTy/zmk-config-klorv2/commit/c59a23e6de66615e96ecabff492bb9ac08902181) seems to be involved, I'm not sure what the goal of renaming was.

You end up with multiple nodes using that `&spi1_default` label, and my guess is that you don't want that unless one is going to replace the other (if this was working before maybe that relied on the node names matching but I haven't had to deal with overlays for the actual controller). I'd try reverting that commit for starters. If that doesn't do it or else you _need_ to modify that node like this you may want to get some guidance in the ZMK Discord, but this isn't anything that conflicts with the keymap editor app.

**Reply by SimonSaysTy** [reply]
Thank you very much, I see what you are talking about and was able to fix it! My apologies that this was not a keymap editor question, I am still learning how to do this all and I appreciate the help greatly!

**Reply by nickcoutsos** [reply]
No worries, I'm glad there was an easy fix and that I sort of understand these things :P
